app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -145,12 +147,12 @@ protected void onCreate(Bundle savedInstanceState) {
         npTrainerLevel.setWrapSelectorWheel(false);
         npTrainerLevel.setValue(trainerLevel);
 
-        Button launch = (Button) findViewById(R.id.start);
-        launch.setOnClickListener(new View.OnClickListener() {
+        launchButton = (Button) findViewById(R.id.start);
+        launchButton.setOnClickListener(new View.OnClickListener() {
 
             @Override
             public void onClick(View v) {
-                if (((Button) v).getText().toString().equals(getString(R.string.main_permission))) {
+                if (!hasAllPermissions()) {
```

Comment:
Yeah, not a strong opinion (this isn't C after all), but if you don't mind do this for the new code.